pytezos/contract/interface.py

```diff
@@ -200,12 +210,47 @@ def storage(self) -> ContractData:
             storage.attach_context(self.context)
         else:
             storage = self.program.storage.dummy(self.context)
-        return ContractData(self.context, storage.item, title='storage')
+        return ContractData(self.context, storage.item, title="storage")
+
+    @cached_property
+    def metadata(self) -> Optional[ContractMetadata]:
+        metadata_url = self.metadata_url
+        if metadata_url is None:
+            return None
+        logger.info('Trying to fetch contract metadata from `%s`', metadata_url)
+        parsed_url = urlparse(metadata_url)
+        metadata_json = {}
+
+        if parsed_url.scheme in ('http', 'https'):
+            # NOTE: KT1B34qXVRfQrScEaqjjt6cJ5G8LtVFZ7fSc
+            metadata_json = self._get_metadata_json_from_http(metadata_url)
```

Comment:
Maybe introduce `ContractMetadata.from_https, ContractMetadata.from_ipfs, ContractMetadata.from_storage` instead? That way one can play with ContractMetadata alone